<launch>
<!-- Launch file for Mapping -->

	<arg name="rviz" default="true" />

    <!-- <rosparam file="$(find fast_lio_sam)/config/avia.yaml" command="load" /> -->
    <rosparam file="$(find fast_lio_sam)/config/mid360.yaml" command="load" />
    <!-- <rosparam file="$(find fast_lio_sam)/config/test_mid360.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/horizon.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/velodyne.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/ouster64.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/vanji720_16.yaml" command="load" /> -->

    <!-- <rosparam file="$(find fast_lio_sam)/config/mapping_backup.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/mapping_dev.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/mapping_imu_backshaft.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/mapping_pointlio.yaml" command="load" /> -->

    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/nclt.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/liosam.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/Hilti2021_ouster.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/Hilti2021_livox.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/Hilti2022.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/UrbanLoco_HK.yaml" command="load" /> -->
    <!-- <rosparam file="$(find fast_lio_sam)/config/dataset/UrbanLoco_CA.yaml" command="load" /> -->

	<param name="showOptimizedPose" type="bool" value="true" />
	<param name="globalMapVisualizationSearchRadius" type="double" value="1000" />
	<param name="globalMapVisualizationPoseDensity" type="double" value="3" />
	<param name="globalMapVisualizationLeafSize" type="double" value="0.2" />

    <node pkg="fast_lio_sam" type="fastlio_sam_ros1" name="system" output="screen" /> 

	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_sam)/rviz_cfg/mapping.rviz" />
	</group>

</launch>